source/common/http/http1/codec_impl.cc

```diff
@@ -474,10 +475,15 @@ void ConnectionImpl::dispatch(Buffer::Instance& data) {
   if (data.length() > 0) {
     for (const Buffer::RawSlice& slice : data.getRawSlices()) {
       total_parsed += dispatchSlice(static_cast<const char*>(slice.mem_), slice.len_);
+      if (HTTP_PARSER_ERRNO(&parser_) != HPE_OK) {
```

Comment:
do we have tests for garbled chunks which would make sure we don't dispatch buffered body when we fail parsing mid-dispatch?  Or do we dispatch given the break rather than return?  I'd think we would want to halt work.

Reply:
The parser can only be in 2 states when we hit this statement: HPE_OK or HPE_PAUSED.
The reason is that other error codes throw an exception in ConnectionImpl::dispatchSlice which unrolls the stack to the HCM.

I had written an invalid chunk headers test to cover this, but it seems to have disappeared due to a bad merge...  See codec_impl_test.cc Http1ServerConnectionImplTest.InvalidChunkHeader 

Thanks for catching this!